src/backend/vulkan/src/lib.rs

```diff
@@ -649,6 +662,54 @@ impl hal::Instance<Backend> for Instance {
     }
 }
 
+unsafe fn load_renderdoc_entrypoint() -> Result<RenderDocEntry, String> {
+    if cfg!(debug_assertions) {
+        {
+            type GetApiFn =
+                unsafe extern "C" fn(version: u32, out: *mut *mut std::ffi::c_void) -> i32;
+
+            #[cfg(windows)]
+            let renderdoc_filename = "renderdoc.dll";
+            #[cfg(all(unix, not(target_os = "android")))]
+            let renderdoc_filename = "librenderdoc.so";
+            #[cfg(target_os = "android")]
+            let renderdoc_filename = "libVkLayer_GLES_RenderDoc.so";
+
+            let renderdoc_lib = libloading::Library::new(renderdoc_filename).map_err(|e| {
+                format!(
+                    "Unable to load renderdoc library '{}': {:?}",
+                    renderdoc_filename, e
+                )
+            })?;
+
+            let get_api: libloading::Symbol<GetApiFn> =
+                renderdoc_lib.get(b"RENDERDOC_GetAPI\0").map_err(|e| {
+                    format!(
+                        "Unable to get RENDERDOC_GetAPI from renderdoc library '{}': {:?}",
+                        renderdoc_filename, e
+                    )
+                })?;
+            let mut obj = std::ptr::null_mut();
+            match get_api(10401, &mut obj) {
+                1 => Ok(RenderDocEntry {
+                    api: *(obj as *mut renderdoc_sys::RENDERDOC_API_1_4_1),
+                    lib: renderdoc_lib,
+                }),
+                return_value => Err(format!(
+                    "Unable to get API from renderdoc library '{}': {}",
+                    renderdoc_filename, return_value
+                )),
+            }
+            .map_err(|error| {
+                warn!("Error loading renderdoc library: {}", error);
+                error
+            })
+        }
+    } else {
+        Err("RenderDoc support is only enabled with 'debug_assertions'".into())
```

Comment:
nit: could early exit here in the beginning of the function, to not indent the whole thing. We have early-outs anyway with "?" in the body.

Reply:
sounds good